### PR TITLE
feat(appconfig-alpha): add deploy method to configuration constructs

### PR DESCRIPTION
**Pull request by chenjane-dev**
Adding a deploy method to the configuration construct to allow deployments on a config at any time after creation.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appconfig-alpha/README.md [hunk 1]
```diff
@@ -176,6 +176,20 @@ new appconfig.HostedConfiguration(this, 'MyHostedConfiguration', {
 });
 ```
 
+To deploy a configuration to an environment after initialization use the `deployTo` method:
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-appconfig-alpha/README.md:179]:
```suggestion
To deploy a configuration to an environment after initialization use the `deploy` method:
```

File: packages/@aws-cdk/aws-appconfig-alpha/test/configuration.test.ts [hunk 2]
```diff
@@ -149,6 +149,121 @@ describe('configuration', () => {
     Template.fromStack(stack).resourceCountIs('AWS::AppConfig::Deployment', 1);
   });
 
+  test('configuration using deploy method and no environment associated', () => {
+    const stack = new cdk.Stack();
+    const app = new Application(stack, 'MyAppConfig', {
+      name: 'MyApplication',
+    });
+    app.addEnvironment('MyEnv1');
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-appconfig-alpha/test/configuration.test.ts:157]:
```suggestion
```

> no environment associated

Shouldn't this be removed?

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks!
Just some minor nits. 
Also, can you please close https://github.com/aws/aws-cdk/pull/28174 in favor of this?

File: packages/@aws-cdk/aws-appconfig-alpha/test/configuration.test.ts (hunk 2 above)
Comment (chenjane-dev) [reply packages/@aws-cdk/aws-appconfig-alpha/test/configuration.test.ts:157]: Those are being attached to the application and not the configuration :)

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Review by vinayak-kukreja — APPROVED** [review]
Hey @chenjane-dev , thank you for this contribution.

**Pull request merged**